presto-main/src/main/java/com/facebook/presto/execution/AccessControlCheckerExecution.java

```diff
@@ -262,6 +270,10 @@ private ListenableFuture<?> executeTask()
         stateMachine.beginColumnAccessPermissionChecking();
         checkAccessPermissions(queryAnalysis.getAccessControlReferences());
         stateMachine.endColumnAccessPermissionChecking();
+        if (preparedQuery.isExplainTypeIO()) {
+            explainIOResult = ((QuerySpecification) ((Query) ((BuiltInQueryAnalysis) queryAnalysis).getAnalysis().getStatement()).getQueryBody()).getFrom()
+                    .map(r -> (((StringLiteral) ((Row) (((Values) ((AliasedRelation) r).getRelation()).getRows().get(0))).getItems().get(0)).getValue()));
```

Comment:
This looks too hacky. Is there a simpler way ?

Reply:
yeah seems hacky. did you test with all IO format types?